File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        

        List<List<Integer>> ans = new ArrayList<>(); 
        if( root == null) return ans; 
        Queue<TreeNode> queue = new LinkedList<>(); 
        queue.add(root) ; 
        while( !queue.isEmpty()) { 
            List<Integer> dep = new ArrayList<>(); 
            int cnt = queue.size() ; 
            for(int i=0 ; i< cnt ; i++){
                TreeNode node = queue.poll() ; 
                dep.add(node.val) ; 
                if( node.left != null) queue.add(node.left) ; 
                if( node.right!= null) queue.add(node.right) ; 
            }
            ans.add(dep) ;
        }
        return ans ; 
        
        
    }
}

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:     
    vector<vector<int>> ans;

    void dfs(TreeNode *node, int dep){
        if(node == NULL) return ; 
        if( ans.size() == dep) {
            ans.push_back( vector<int>()); } 

        ans[dep].push_back(node->val); 
        dfs(node->left , dep+1) ; 
        dfs(node->right, dep+1) ; 
    }

    vector<vector<int>> levelOrder(TreeNode* root) {

        dfs(root,0);
        return ans ; 

        
    }
};